### PR TITLE
MGMT-7365: Remove single network configuration fields

### PR DESCRIPTION
**Pull request by mkowalski**
During the dual-stack implementation the new structures have been
introduced - machine_networks, cluster_networks and service_networks
(all of them of type []string). In order to preserve backwards
compatibility, there were wrappers created that translate e.g.
machine_network_cidr (of type string) to machine_networks (of type
[]string).

Usage of the old parameters makes it impossible to pass multiple
networks (because of type mismatch). Given that in API v2 the new
structures ([]string) are fully implemented, there is no need to support
old ones any more.

Closes: [MGMT-7365](https://issues.redhat.com/browse/MGMT-7365)

## List all the issues related to this PR

- [ ] New Feature 
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @carbonin 
/cc @nmagnezi 

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/migrations/20210822134659_multiple_networks.go [hunk 1]
```diff
@@ -1,66 +0,0 @@
-package migrations
-
-import (
-	gormigrate "github.com/go-gormigrate/gormigrate/v2"
-	"github.com/openshift/assisted-service/internal/common"
-	"github.com/openshift/assisted-service/models"
-	"gorm.io/gorm"
-)
-
-func multipleNetworks() *gormigrate.Migration {
```
Comment (carbonin) [inline internal/migrations/20210822134659_multiple_networks.go:10]: So theoretically we should keep this migration.
Could there be some installation out there that still hasn't run this migration, but will still want to move the cluster data from the old columns to the new tables when they get around to upgrading?

Specifically I wonder if ACM 2.3 is still supported and if people can upgrade directly from 2.3 to 2.6 or something.
I don't know if this migration pre-dates 2.3, but the idea is that it should work effectively forever.

File: internal/migrations/20220324203547_multiple_networks_cleanup.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		err = tx.Migrator().DropColumn(&clusters, "service_network_cidr")
+		if err != nil {
+			log.Printf("Non-critical error: %s", err)
+		}
+
+		return nil
+	}
+
+	rollback := func(tx *gorm.DB) error {
+		// As the migration never returns any errors, we are not implementing rollback.
```
Comment (carbonin) [inline internal/migrations/20220324203547_multiple_networks_cleanup.go:37]: So again, theoretically this should add the columns back, but if we need to rollback we'd probably deploy an image earlier than this change which should automigrate the columns back into existence, right?

**Comment by mkowalski** [comment]
/hold
Will redesign so that we don't remove columns from schema

**Pull request closed without merging**